openproblems/api/hash.py

```diff
@@ -32,12 +32,17 @@ def get_module(fun):
     return fun.__module__
 
 
-def git_hash(file):
+def git_hash(obj):
     """Get the git commit hash associated with a file."""
-    return _run(
-        ["git", "log", "-n", "1", "--pretty=format:%H", "--", file],
-        cwd=os.path.dirname(__file__),
-    )
+    if isinstance(obj, str) and os.path.isfile(obj):
+        return _run(
+            ["git", "log", "-n", "1", "--pretty=format:%H", "--", obj],
+            cwd=os.path.dirname(__file__),
+        )
+    elif hasattr(obj, "__file__"):
+        return git_hash(obj.__file__)
+    elif callable(obj):
+        return git_hash(importlib.import_module(get_module(obj)))
```

Comment:
How does this function work? Could you add some description so I know how to replicate its behaviour if need be?

Reply:
It's just the latest commit which modified the file / function / submodule. Added some comments.